packages/workflow/src/workflow.ts

```diff
@@ -573,25 +638,38 @@ export function proxyActivities<A = UntypedActivities>(options: ActivityOptions)
  *
  * @see {@link proxyActivities} for examples
  */
-export function proxyLocalActivities<A = UntypedActivities>(options: LocalActivityOptions): ActivityInterfaceFor<A> {
+export function proxyLocalActivities<A = UntypedActivities>(
+  options: LocalActivityOptions
+): ActivityInterfaceWithOptions<A> {
   if (options === undefined) {
     throw new TypeError('options must be defined');
   }
   // Validate as early as possible for immediate user feedback
   validateLocalActivityOptions(options);
-  return new Proxy(
-    {},
-    {
+
+  function createActivityProxy(options: ActivityOptions): ActivityInterfaceWithOptions<A> {
+    return new Proxy({} as ActivityInterfaceWithOptions<A>, {
       get(_, activityType) {
         if (typeof activityType !== 'string') {
           throw new TypeError(`Only strings are supported for Activity types, got: ${String(activityType)}`);
         }
-        return function localActivityProxyFunction(...args: unknown[]) {
-          return scheduleLocalActivity(activityType, args, options);
+
+        function activityProxyFunction(...args: unknown[]): Promise<unknown> {
```

Comment:
That function name may be visible in stack traces on the workflow side. Better keep the "localActivity" name

Reply:
reverted name